spec/controllers/projects_controller_spec.rb

```diff
@@ -186,6 +186,31 @@
     end
   end
 
+  describe "GET permalink_valid?" do
+    context "when the permalink is already taken" do
+      let(:permalink) { "peramlink_taken" }
```

Comment:
Wouldn't be better to extract this let to a higher level? So you won't have to instantiate the permalink variable twice.